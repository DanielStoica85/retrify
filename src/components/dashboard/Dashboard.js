import React, { Component } from 'react';
import { Container, Row, Col } from 'reactstrap';
import Notifications from './Notifications';
import RetrosList from '../retros/RetrosList';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export class Dashboard extends Component {
    render() {
        return (
            <Container className="dashboard">
                <Row>
                    <Col sm="12" md="6">
                        <RetrosList retros={this.props.retros} />
                    </Col>
                    <Col sm="12" md={{ size: 5, offset: 1 }}>
                        <Notifications />
                    </Col>
                </Row>
            </Container>
        );
    }
}

const mapStateToProps = state => {
    return {
        retros: state.retros
    };
};

Dashboard.propTypes = {
    retros: PropTypes.array
};

export default connect(mapStateToProps)(Dashboard);
