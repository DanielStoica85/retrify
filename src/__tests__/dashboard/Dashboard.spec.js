import React from 'react';
import renderer from 'react-test-renderer';
import Dashboard from '../../components/dashboard/Dashboard';

describe('Dashboard component', () => {
    it('renders correctly', () => {
        const component = renderer.create(<Dashboard />).toJSON();
        expect(component).toMatchSnapshot();
    });
});
