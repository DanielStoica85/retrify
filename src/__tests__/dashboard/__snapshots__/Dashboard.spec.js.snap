// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component renders correctly 1`] = `
<div
  className="dashboard container"
>
  <div
    className="row"
  >
    <div
      className="col-sm-12 col-md-6"
    >
      <div
        className="retros-list section"
      >
        <div>
          <div
            className="mt-3 card"
          >
            <div
              className="card-body"
            >
              <div
                className="card-title"
              >
                Some title
              </div>
              <div
                className="card-subtitle"
              >
                Added by 
                Some author
              </div>
            </div>
            <div
              className="card-footer"
              style={
                Object {
                  "backgroundColor": "lightgray",
                  "borderColor": "lightgray",
                }
              }
            >
              May 10th, 2019
            </div>
          </div>
        </div>
        <div>
          <div
            className="mt-3 card"
          >
            <div
              className="card-body"
            >
              <div
                className="card-title"
              >
                Some title
              </div>
              <div
                className="card-subtitle"
              >
                Added by 
                Some author
              </div>
            </div>
            <div
              className="card-footer"
              style={
                Object {
                  "backgroundColor": "lightgray",
                  "borderColor": "lightgray",
                }
              }
            >
              May 10th, 2019
            </div>
          </div>
        </div>
        <div>
          <div
            className="mt-3 card"
          >
            <div
              className="card-body"
            >
              <div
                className="card-title"
              >
                Some title
              </div>
              <div
                className="card-subtitle"
              >
                Added by 
                Some author
              </div>
            </div>
            <div
              className="card-footer"
              style={
                Object {
                  "backgroundColor": "lightgray",
                  "borderColor": "lightgray",
                }
              }
            >
              May 10th, 2019
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-sm-12 col-md-5 offset-md-1"
    >
      <div>
        <p>
          Notifications
        </p>
      </div>
    </div>
  </div>
</div>
`;
